### PR TITLE
Add gen mod for more convenient to use

### PR DESCRIPTION
**Pull request by jokemanfire**
while use customize ,you can use gen_mod to generate mod.rs.

### CODE REVIEW

**Comment by Tim-Zhang** [comment]
Hi @jokemanfire, I think it would be better if you can add some UT or directly use gen_mod in the example. thanks

**Comment by Tim-Zhang** [comment]
BTW, The CI failed

File: compiler/src/codegen.rs [hunk 1]
```diff
@@ -724,6 +724,30 @@ pub fn gen_and_write(
...
+        //if mod file exists
+        if let Ok(file) = File::open(&file_path){
+            let reader = io::BufReader::new(file);
+            reader.lines().for_each(|line|{
+               if let Ok(r) = line{
+                    set.insert(r);
+               }  
+            });
+        }
+        let mut file_write = fs::OpenOptions::new().write(true).truncate(true).open(&file_path)?;
```
Comment (Tim-Zhang) [inline compiler/src/codegen.rs:740]: needs `.create(true)` here when mod file does not exist

File: compiler/src/codegen.rs [hunk 2]
```diff
@@ -724,6 +724,30 @@ pub fn gen_and_write(
...
+        for r in &results {
+            let prefix_name:Vec<&str> = r.name.split(".").collect();
+            set.insert(format!("pub mod {};", prefix_name[0]));
```
Comment (Tim-Zhang) [inline compiler/src/codegen.rs:743]: It's not enough to add `pub mod *_ttrpc` here, still we need `pub mod *`

**Comment by jokemanfire** [comment]
> Hi @jokemanfire, I think it would be better if you can add some UT or directly use gen_mod in the example. thanks

OK I will change it.

**Comment by Tim-Zhang** [comment]
@jokemanfire I have formatted the code and have deleted `mod.rs` of example from repo

**Review by Tim-Zhang — APPROVED** [review]
LGTM, thanks @jokemanfire

**Review by lifupan — APPROVED** [review]

**Pull request merged**